web/src/components/forms/access-rule/steps/Time.tsx

```diff
@@ -1,14 +1,22 @@
 import { FormControl, FormLabel, Text, VStack } from "@chakra-ui/react";
 import React from "react";
 import { Controller, useFormContext } from "react-hook-form";
-import { DurationInput, Hours, Minutes } from "../../../DurationInput";
+import { durationString } from "../../../../utils/durationString";
+import {
+  Days,
+  DurationInput,
+  Hours,
+  Minutes,
+  Months,
+  Weeks,
+} from "../../../DurationInput";
 import { AccessRuleFormData } from "../CreateForm";
 import { FormStep } from "./FormStep";
 
 export const TimeStep: React.FC = () => {
   const methods = useFormContext<AccessRuleFormData>();
   const time = methods.watch("timeConstraints");
-  const maxDurationSeconds = 24 * 3600;
+  const maxDurationSeconds = 6 * 4 * 7 * 24 * 3600; // 6 months in seconds
```

Comment:
This should be the correct calculation 
```suggestion
  const maxDurationSeconds = 26 * 24 * 60 * 60; // 6 months in seconds = 2246400
```